OrbitVulkanLayer/SubmissionTracker.h

```diff
@@ -0,0 +1,816 @@
+ public:
+  // On a submission (vkQueueSubmit), all pointers to command buffers become invalid/can be reused
+  // for0a the next submission. The struct `QueueSubmission` gathers information (like timestamps
+  // and timer slots) about a concrete submission and their corresponding command buffers and debug
+  // markers. This way the information is "persistent" across submissions. We will create this
+  // struct at `vkQueuePresent` (right before the call into the driver -- and only if we are
+  // capturing) and add some information (in particular markers and a timestamp) right after the the
+  // driver call. So, we will use `QueueSubmission` as a return value, and therefore these structs
+  // need to be public in the `SubmissionTracker`.
+
+  // Stores meta information about a submit (VkQueueSubmit), e.g. to allow to identify the matching
+  // Gpu tracepoint.
+  struct SubmissionMetaInformation {
+    uint64_t pre_submission_cpu_timestamp;
+    uint64_t post_submission_cpu_timestamp;
+    int32_t thread_id;
+  };
+
+  // A persistent version of a command buffer that was submitted and its begin/end slot in the
+  // `TimerQueryPool`. Note that the begin is optional, as it might not be part of the capture.
+  // This struct is created if we capture the submission, so if we haven't captured the end,
+  // we also haven't captured the begin, and don't need to store info about that command buffer at
+  // all. The list of all `SubmittedCommandBuffer`s is stored in a `QueueSubmission`, and can be so
+  // associated to the `SubmissionMetaInformation`.
+  struct SubmittedCommandBuffer {
+    std::optional<uint32_t> command_buffer_begin_slot_index;
+    uint32_t command_buffer_end_slot_index;
+  };
+
+  // A persistent version of a debug marker (either begin or end) that was submitted and its slot
+  // in the `TimerQueryPool`. `SubmittedMarkers` are used in `MarkerState` to identify the begin or
+  // end marker. All markers (`MarkerState`s) that gets *completed* within a certain submission are
+  // stored in the `QueueSubmission`.
+  // Note that we also store the `SubmissionMetaInformation` as begin markers
+  // can originate from different submissions than the matching end markers. This allows us e.g. to
+  // match the markers to a specific submission tracepoint. So even though the submitted markers a
+  struct SubmittedMarker {
+    SubmissionMetaInformation meta_information;
+    uint32_t slot_index;
+  };
+
+  // Represents a color to be used to in debug markers. The values are all in range [0.f, 1.f].
+  struct Color {
+    float red;
+    float green;
+    float blue;
+    float alpha;
+  };
+
+  // Identifies a particular debug marker region that has been submitted via `vkQueueSubmit`.
+  // Note that we only store the state into `QueueSubmission`, if at that time we have a value for
+  // the end_info. Beside the information about the begin/end, it also stores the text, color and
```

Comment:
Update comment to reflect name change from "text" to "label_name"?

Reply:
Done.